Wikipedia/Code/MWKSearchResult.m

```diff
@@ -235,4 +235,33 @@ - (nullable NSURL *)articleURLForSiteURL:(NSURL *)siteURL {
     return [siteURL wmf_URLWithTitle:self.displayTitle];
 }
 
+#pragma mark - NSObject
+
+- (BOOL)isEqualToSearchResult:(MWKSearchResult *)searchResult {
+    if (!searchResult) {
+        return NO;
+    }
+    
+    BOOL haveEqualDisplayTitles = (!self.displayTitle && !searchResult.displayTitle) || [self.displayTitle isEqualToString:searchResult.displayTitle];
+    
+    return haveEqualDisplayTitles;
+}
+
+- (BOOL)isEqual:(id)object {
+    if (self == object) {
+        return YES;
+    }
+    
+    if (![object isKindOfClass:[MWKSearchResult class]]) {
+        return NO;
+    }
+    
+    return [self isEqualToSearchResult:(MWKSearchResult *)object];
+    
+}
+
+- (NSUInteger)hash {
+    return [self.displayTitle hash];
+}
```

Comment:
@nambatee I thought about the same solution as yours, but I don't think that overriding `isEqual:` and `hash` methods is the right way to correct this. MWKSearchResults is a subclass of `MTLModel` that checks equality of two `MTLModel` objects not based on their adress, but on `MTLModel` implementation which uses propertie keyx defined inthe objects. 

So, I think the problem is that one of the property doesn't return `true` when `isEqual:` is called on it, while it should.